docs/concepts/abstractions/controllers/garbage-collection.md

```diff
@@ -50,59 +53,91 @@ metadata:
   ownerReferences:
   - apiVersion: extensions/v1beta1
     controller: true
+    blockOwnerDeletion: true
     kind: ReplicaSet
     name: my-repset
     uid: d9607e19-f88f-11e6-a518-42010a800195
   ...
 ```
 
-## Controlling whether the garbage collector deletes dependents
+## Controlling whether and how the garbage collector deletes dependents
+
+When you delete object, you can specify whether the object's dependents are
+deleted automatically. If you delete an object without deleting its dependents
+automatically, the dependents are said to be *orphaned*. Deleting dependents
+automatically is called *cascading deletion*. Further, there are two modes of
+*cascading deletion*: if the object is deleted immediately and the garbage
+collector then deletes the dependents in the background, it is called
+*background cascading deletion*. In contrast, in *foreground cascading
+deletion*, the object first enters a "deletion in progress" state, where the
+object is still visible via the REST API, its `deletionTimestamp` is set, and
+its metadata.finalizers contains "foregroundDeletion". Then the garbage
+collector deletes the dependents. Once the garbage collector has deleted all
+dependents whose ownerReferences.blockOwnerDeletion=true, it will finally delete
+the object.
+
+To control whether delete dependent objects or delete them in
+foreground/background, set the deleteOptions.propagationPolicy to "Orphan",
+"Foregound", or "Background" respectively.
+
+The default garbage collection policy for controller resources is `orphan`. So unless you specify
```

Comment:
I should scope it to "controllers that automatically sets ownerReferences for controllees"